cmd/ghz/main.go

```diff
@@ -315,11 +315,34 @@ func createConfigFromArgs(cfg *runner.Config) error {
 		binaryData = b
 	}
 
-	var metadata map[string]string
+	var metadataArray []map[string]string
+	var metadataMap map[string]string
+
 	*md = strings.TrimSpace(*md)
 	if *md != "" {
-		if err := json.Unmarshal([]byte(*md), &metadata); err != nil {
-			return fmt.Errorf("Error unmarshaling metadata '%v': %v", *md, err.Error())
+		// For backward compatibility reasons we support both approaches - specifying an array
+		// with multiple object items and specifying a single object
+
+		// 1. First try de-serializing it into an object
+		if err := json.Unmarshal([]byte(*md), &metadataMap); err != nil {
+			if !strings.Contains(err.Error(), "cannot unmarshal array into Go value of type map") {
+				// Some other fatal error which we should immediately propagate instead of try to
+				// de-serializing input into an array of maps (e.g. unexpected end of JSON input,
+				//etc.)
+				return fmt.Errorf("Error unmarshaling metadata '%v': %v", *md, err.Error())
+			}
+
+			// 2. If that  fails, try to de-serialize it into an array
+			// NOTE: We could also simply check if string begins with [ or {, but that approach is
+			// not 100% robust
+
+			if err := json.Unmarshal([]byte(*md), &metadataArray); err != nil {
+				return fmt.Errorf("Error unmarshaling metadata '%v': %v", *md, err.Error())
+			}
+		}
```

Comment:
With respect to determining the underlying type we are trying to Unmarshal, instead of relying on `strings.Contains()` of the error message, I think we can take advantage of the actual [UnmarshalTypeError](https://golang.org/pkg/encoding/json/#UnmarshalTypeError). The approach may be a little bit more reliable. Example of what I mean:

```go
md := `["foo", "bar"]`
	var metadataMap map[string]string
	if err := json.Unmarshal([]byte(md), &metadataMap); err != nil {
		if e, ok := err.(*json.UnmarshalTypeError); ok && e.Value == "array" {
			fmt.Println("trying to Unmarshal array into map", e)
		}
	} else {
		fmt.Println(metadataMap)
	}
```